install/test/helm_test.go

```diff
@@ -5884,6 +5887,38 @@ metadata:
 							if u.GetKind() == kind && u.GetName() == resourceName {
 								a := getFieldFromUnstructured(u, append(prefixPath, "spec", "ttlSecondsAfterFinished")...)
 								Expect(a).To(Equal(int64(TTL_SECONDS_AFTER_FINISHED)))
+								if helmHookDeletePolicy, ok := u.GetAnnotations()["helm.sh/hook-delete-policy"]; ok {
+									Expect(helmHookDeletePolicy).NotTo(ContainSubstring("hook-succeeded"))
+								}
```

Comment:
```suggestion
								if helmHookDeletePolicy, ok := u.GetAnnotations()["helm.sh/hook-delete-								if kind == "Job" {
									helmHookDeletePolicy, ok := u.GetAnnotations()["helm.sh/hook-delete-policy"]
									if resourceName == "gateway-certgen" {
										Expect(ok).To(BeTrue())
										Expect(helmHookDeletePolicy).NotTo(ContainSubstring("hook-succeeded"))
									} else if resourceName == "gloo-mtls-certgen" {
										Expect(ok).To(BeFalse())
									}
								}
```

As above, we can be more explicit about when we expect `hook-delete-policy` to be present